smart-client-core/src/main/java/com/emc/rest/smart/SmartConfig.java

```diff
@@ -132,6 +132,12 @@ public Object getProperty(String propName) {
         return properties.get(propName);
     }
 
+    public Integer getPropertyIntValue(Object propertyValue, int defaultValue) {
+        String pValue = (String)propertyValue;
+        boolean isNumeric = pValue != null && pValue.matches("[0-9.]+");
```

Comment:
Note that integers should not have a decimal - this regex will also match "1.2.3..4" and "...", which are not numbers.

You might want to just call `Integer.parseInt()` in a try block, and return the default if any exception is thrown.